src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp

```diff
@@ -42,6 +42,10 @@ bool canUseProjectionForReadingStep(ReadFromMergeTree * reading)
     if (reading->getContext()->getSettingsRef().allow_experimental_query_deduplication)
         return false;
 
+    // Currently projection don't support settings which implicitly modify aggregate functions.
+    if (reading->getContext()->getSettingsRef().aggregate_functions_null_for_empty)
+        return false;
+
```

Comment:
Maybe also add this check to [old code](https://github.com/ClickHouse/ClickHouse/blob/25912a267311fc5aaa275edf6d72c2e1e0f97d34/src/Storages/MergeTree/MergeTreeData.cpp#L6606) for projections analysis?

Reply:
It's already there. See  #42198

Reply:
Ah, sorry. Missed that.